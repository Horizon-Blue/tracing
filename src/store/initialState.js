export default {
  locale: 'en',
  token: null,
};
