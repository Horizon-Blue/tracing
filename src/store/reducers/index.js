import { combineReducers } from 'redux';
import locale from './locale';
import token from './token';

export default combineReducers({ locale, token });
