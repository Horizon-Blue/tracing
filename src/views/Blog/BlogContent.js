import React, { PureComponent } from 'react';
import markdownIt from 'markdown-it';
import mk from 'markdown-it-katex';
import prism from 'markdown-it-prism';
import { Container } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import './prism-atom-dark.css';

const md = markdownIt({ html: true, typographer: true, xhtmlOut: true })
  .use(mk)
  .use(prism);

/**
 * The component that renders blog post content (Markdown string) into
 * DOM nodes
 *
 * @class      BlogContent (name)
 */
class BlogContent extends PureComponent {
  static propTypes = {
    content: PropTypes.string.isRequired,
  };

  /**
   * Helper function to render content
   *
   * @param      {<type>}  text    The text
   * @return     {<type>}  { description_of_the_return_value }
   */
  static renderPost(text) {
    return md.render(text);
  }

  render() {
    return (
      <Container text textAlign="left">
        <main
          dangerouslySetInnerHTML={{
            __html: BlogContent.renderPost(this.props.content),
          }}
        />
      </Container>
    );
  }
}

export default BlogContent;
