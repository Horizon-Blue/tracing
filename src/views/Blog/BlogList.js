import React, { PureComponent } from 'react';
import { Container } from 'semantic-ui-react';
import { Seq } from 'immutable';
import PropTypes from 'prop-types';
import BlogListItem from './BlogListItem';
import gql from 'graphql-tag';
import { withRouter } from 'react-router-dom';
import { graphql } from 'react-apollo';
import Loading from '../Loading';
import ErrorMessage from '../ErrorMessage';

/**
 * The content to shows on the blog list page
 *
 * @class      BlogList (name)
 */
class BlogList extends PureComponent {
  static propTypes = {
    data: PropTypes.object.isRequired,
    query: PropTypes.string,
  };

  /**
   * Given each post, render it as an entry in the blog list
   *
   * @param      {object}  post    The post
   * @return     {Node}  the rendered React component
   */
  renderPostItem({ node: post }) {
    return <BlogListItem key={post.id} post={post} />;
  }

  render() {
    const { data: { loading, posts, error } } = this.props;

    if (loading) return <Loading />;

    if (error) return <ErrorMessage value={error} />;

    return (
      <Container as="main">
        {Seq(posts.edges)
          .reverse()
          .map(this.renderPostItem)}
      </Container>
    );
  }
}

const postField = `
edges {
  node {
    id
    title
    excerpt
    category {
      id
      name
    }
    tags {
      edges {
        node {
          id
          name
        }
      }
    }
    author {
      id
      name
    }
    publishDate
  }
}`;

const getPostWithTag = gql`
query ($tagId: ID!){
  tag(id: $tagId) {
    posts {
      ${postField}
    }
  }
}
`;

const getPostWithCategory = gql`
query ($categoryId: ID!){
  category(id: $categoryId) {
    posts {
      ${postField}
    }
  }
}
`;

const getAllPosts = gql`
  query($query: String) {
    posts(query: $query) {
      ${postField}
    }
  }
`;

export { BlogList as BlogListView };
export default withRouter(
  graphql(getPostWithCategory, {
    skip: ({ match: { params: { categoryId } } }) => !categoryId,
    options: ({ match: { params: { categoryId } } }) => ({
      variables: { categoryId },
    }),
    props: ({ data: { category, ...restData }, ...rest }) => ({
      ...rest,
      data: {
        ...restData,
        posts: category ? category.posts : { edges: [] },
      },
    }),
  })(
    graphql(getPostWithTag, {
      skip: ({ match: { params: { tagId } } }) => !tagId,
      options: ({ match: { params: { tagId } } }) => ({
        variables: { tagId },
      }),
      props: ({ data: { tag, ...restData }, ...rest }) => ({
        ...rest,
        data: {
          ...restData,
          posts: tag ? tag.posts : { edges: [] },
        },
      }),
    })(
      graphql(getAllPosts, {
        skip: ({ match: { params: { tagId, categoryId } } }) =>
          tagId || categoryId,
        options: ({ query }) => ({
          variables: { query },
        }),
      })(BlogList)
    )
  )
);
