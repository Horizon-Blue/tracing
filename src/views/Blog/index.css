section.blog-list {
  padding: 5vh 5vw;
  border-bottom: 1px solid rgb(35, 36, 45);
  transition: background-color 500ms ease-in, color 500ms ease-in;
}

section.blog-list:hover {
  background-color: rgba(12, 96, 170, 0.3);
}

.ui.basic.blue.label {
  background-color: rgba(14, 42, 118, 0.4) !important;
  font-size: 0.9em;
}

section.blog-list .tag-container {
  margin-top: 1rem !important;
}

@media only screen and (min-width: 992px) {
  section.blog-list .tag-container {
    text-align: right;
    margin-top: 0 !important;
  }
}

section.blog-list summary {
  color: rgba(157, 204, 254, 0.8);
  font-size: 1.1em;
}

section.blog-list .blog-meta-divider {
  margin: 0 0.5rem;
}

article.blog-post {
  padding-top: 3vh;
  color: white;
}

article.blog-post .blog-post-meta {
  color: rgba(157, 204, 254, 0.8);
}

article.blog-post .blog-post-meta span:not(:first-child) {
  margin-left: 1vw;
}

article.blog-post main {
  margin: 2vh 0 5vh 0;
}
