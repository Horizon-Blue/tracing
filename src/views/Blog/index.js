import React, { PureComponent } from 'react';
import { Route, Switch } from 'react-router-dom';
import BlogList from './BlogList';
import BlogPost from './BlogPost';
import './index.less';

/**
 * The Blog Section.
 *
 * @class      Blog (name)
 */
class Blog extends PureComponent {
  render() {
    return (
      <Switch>
        <Route exact path="/blog" component={BlogList} />
        <Route path="/blog/:postId" component={BlogPost} />
      </Switch>
    );
  }
}

export default Blog;
