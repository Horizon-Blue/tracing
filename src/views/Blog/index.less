section.blog-list {
  padding: 5vh 5vw;
  border-bottom: 1px solid rgb(35, 36, 45);
  transition: background-color 500ms ease-in, color 500ms ease-in;

  &:hover {
    background-color: rgba(12, 96, 170, 0.3);
  }

  .tag-container {
    margin-top: 1rem !important;
    @media (min-width: 992px) {
      text-align: right;
      margin-top: 0 !important;
    }
  }

  summary {
    color: rgba(157, 204, 254, 0.8);
    font-size: 1.1em;
  }

  .blog-meta-divider {
    margin: 0 0.5rem;
  }
}

.ui.basic.blue.label {
  background-color: rgba(14, 42, 118, 0.4) !important;
  font-size: 0.9em;
}

form.search-box {
  margin-top: 2vh;
  margin-left: 5vw;
  border-bottom: 1px solid rgb(35, 36, 45);
}

article.blog-post {
  padding-top: 3vh;
  color: white;

  h2,
  p,
  .comment-username {
    color: white !important;
  }

  .blog-post-meta,
  .comment-meta,
  .comment-action a {
    color: rgba(157, 204, 254, 0.8) !important;
  }

  .blog-post-meta > span:not(:first-child) {
    margin-left: 1vw;
  }

  main {
    margin: 2vh 0 5vh 0;
  }

  .comment-container {
    margin-bottom: 1vh !important;
    padding: 1vh 1vh !important;
    transition: background-color 500ms ease-in, color 500ms ease-in;

    &:hover {
      background-color: rgba(12, 96, 170, 0.3);
    }
  }
}
