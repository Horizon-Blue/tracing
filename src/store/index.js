import { createStore } from 'redux';
import initialState from './initialState';
import rootReducer from './reducers';

const store = createStore(
  rootReducer,
  initialState,
  /* Enable Redux Devtools Chrome Extension */
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

export default store;
