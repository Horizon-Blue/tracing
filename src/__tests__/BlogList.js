import React from 'react';
import { BlogListView } from '../views/Blog/BlogList';
import renderer from 'react-test-renderer';
import store from '../store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

it('renders loading', () => {
  const tree = renderer
    .create(<BlogListView data={{ loading: true }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

const posts = {
  edges: [
    {
      node: {
        id: 1234,
        title: 'Foo Bar',
        author: {
          name: 'Xiaoyan',
        },
        excerpt: 'This is an excerpt',
        tags: {
          edges: [{ node: { id: 1233, name: '标签' } }],
        },
        category: {
          name: 'journal',
          href: 'journal',
        },
        publishDate: '2017-06-26T23:20:11',
      },
    },
    {
      node: {
        id: 124,
        title: '测试',
        author: {
          name: 'Xiaoyan',
        },
        excerpt:
          'something to test. The quick brown fox jumps over the lazy dog the quick brown fox jumps over the lazy dog.',
        tags: {
          edges: [{ node: { id: 133, name: 'test' } }],
        },
        category: {
          name: 'journal',
          href: 'journal',
        },
        publishDate: '2018-01-04T22:18:25Z',
      },
    },
  ],
};

it('render with content', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <BlogListView data={{ loading: false, posts }} />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
