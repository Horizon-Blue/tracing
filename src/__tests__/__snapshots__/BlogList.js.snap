// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with content 1`] = `
<main
  className="ui container"
>
  <a
    href="/blog/post/1234"
    onClick={[Function]}
  >
    <section
      className="ui container blog-list"
    >
      <div
        className="ui inverted grid"
      >
        <div
          className="row"
        >
          <div
            className="twelve wide computer sixteen wide mobile column"
          >
            <h2
              className="ui inverted header"
            >
              Foo Bar
            </h2>
          </div>
          <div
            className="right floated four wide computer sixteen wide mobile column tag-container"
          >
            <div
              className="ui blue basic horizontal label"
              onClick={[Function]}
            >
              标签
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="column"
          >
            <summary>
              This is an excerpt
            </summary>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="column"
          >
            <div>
              <span
                dispatch={undefined}
              >
                Xiaoyan posted at 2017-06-26
              </span>
              <span
                className="blog-meta-divider"
              >
                |
              </span>
              <i>
                journal
              </i>
            </div>
          </div>
        </div>
      </div>
    </section>
  </a>
  <a
    href="/blog/post/124"
    onClick={[Function]}
  >
    <section
      className="ui container blog-list"
    >
      <div
        className="ui inverted grid"
      >
        <div
          className="row"
        >
          <div
            className="twelve wide computer sixteen wide mobile column"
          >
            <h2
              className="ui inverted header"
            >
              测试
            </h2>
          </div>
          <div
            className="right floated four wide computer sixteen wide mobile column tag-container"
          >
            <div
              className="ui blue basic horizontal label"
              onClick={[Function]}
            >
              test
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="column"
          >
            <summary>
              something to test. The quick brown fox jumps over the lazy dog the quick brown fox jumps over the lazy dog.
            </summary>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="column"
          >
            <div>
              <span
                dispatch={undefined}
              >
                Xiaoyan posted at 2018-01-04
              </span>
              <span
                className="blog-meta-divider"
              >
                |
              </span>
              <i>
                journal
              </i>
            </div>
          </div>
        </div>
      </div>
    </section>
  </a>
</main>
`;

exports[`renders loading 1`] = `
<main
  className="ui container"
>
  <div
    className="ui big active inverted centered inline loader"
  />
</main>
`;
